src/classes/BDI_DataImportService.cls

```diff
@@ -0,0 +1,2252 @@
+                    new string[]{diSettings.Account_Custom_Unique_ID__c, strAccount1, strAccount2})));
+            }
+        }
+
+        // make sure they have specified a valid Donation Matching Behavior
+        set<String> setDMBehavior = new set<String> {
+            BDI_DataImport_API.DoNotMatch,
+            BDI_DataImport_API.RequireNoMatch,
+            BDI_DataImport_API.RequireExactMatch,
+            BDI_DataImport_API.ExactMatchOrCreate,
+            BDI_DataImport_API.RequireBestMatch,
+            BDI_DataImport_API.BestMatchOrCreate
+        };
+        if (!setDMBehavior.contains(diSettings.Donation_Matching_Behavior__c)) {
+            throw(new BDIException(String.format(Label.bdiInvalidDonationMatchingBehavior, new String[]{diSettings.Donation_Matching_Behavior__c})));
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description update the (optional) Batch's statistics.
+    * @param apexJobId the apexJob this process was running under
+    * @param batchId the Id of the batch to update
+    * @return void
+    */
+    public static void updateDIBatchStatistics(Id apexJobId, Id batchId) {
+        if (batchId != null) {
+            DataImportBatch__c diBatch = loadBatch(batchId);
+            integer cRecordsProcessed = [
+                SELECT COUNT()
+                FROM DataImport__c
+                WHERE NPSP_Data_Import_Batch__c = :diBatch.Id
+                AND ApexJobId__c = :apexJobId
+            ];
+            integer cRecordsSucceeded = [
+                SELECT COUNT()
+                FROM DataImport__c
+                WHERE NPSP_Data_Import_Batch__c = :diBatch.Id
+                AND Status__c = :label.bdiImported
+                AND ApexJobId__c = :apexJobId
+            ];
+            integer cRecordsFailed = cRecordsProcessed - cRecordsSucceeded;
+
+            diBatch.Last_Processed_On__c = System.now();
+            diBatch.Records_Successfully_Processed__c = cRecordsSucceeded;
+            diBatch.Records_Failed__c = cRecordsFailed;
+            update diBatch;
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description maps to keep track of the objects we will create.
+    * all are indexed by the object's matching key or the di.Id
+    */
+    public map<string, Contact> mapDIKeyToContact = new map<string, Contact>();
+    public map<string, Account> mapDIKeyToAccount = new map<string, Account>();
+    private map<Id, Address__c> mapDIIdToAddr = new map<Id, Address__c>();
+    public map<string, Id> mapStrNameToCampaignId = new map<string, Id>();
+
+    /*******************************************************************************************************
+    * @description holds the list of DataImport records we are currently processing
+    */
+    global list<DataImport__c> listDI;
```

Comment:
Should these global vars be moved further up so they're more immediately apparent?